tests/lib/modem_trace/modem_trace_rtt/Kconfig

```diff
@@ -0,0 +1,23 @@
+config HAS_SEGGER_RTT
+	bool
+	default y
+	help
+	    This symbol overrides the promptless symbol HAS_SEGGER_RTT because this is needed by
+	    the unit test.
+
+choice NRF_MODEM_LIB_TRACE_MEDIUM
+	prompt "Test: nRF modem trace medium"
+	default NRF_MODEM_LIB_TRACE_MEDIUM_RTT
+	help
+	    This symbols overrides the symbol NRF_MODEM_LIB_TRACE_MEDIUM in order to override
+		the latter's dependencies on NRF_MODEM_LIB and its dependencies.
+endchoice
+
+config NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE
+	int "Test: Size of the buffer used by the RTT to write traces"
+	default 255
+	help
+	    This symbols overrides the symbol NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE in order to
+		override the latter's dependencies on NRF_MODEM_LIB and its dependencies.
```

Comment:
This is already default here:
https://github.com/nrfconnect/sdk-nrf/blob/ece4a34ada95cd842dc843131934e496391292c6/lib/nrf_modem_lib/Kconfig#L64-L66

So if you add the proposed changes to `prj.conf`, then this whole code block can go away.
```suggestion
```

Reply:
alternatives has been discussed offline, work will be carried out to establish some unit test and mock guidelines in general. 

But that is considered outside the scope of this PR, so accepting this for now.

No changes requested.